components/mosart/src/riverroute/RtmMod.F90

```diff
@@ -2352,11 +2397,99 @@ subroutine Rtmrun(rstwr,nlend,rdate)
     endif ! budget_check
 
     ! data for euler solver, in m3/s here
+
+    ! Aggregate global Qgwl (TNR) if flag is true ---
+    global_negative_qgwl_sum = 0.0_r8 ! Reset global sum each step
+
+     if (redirect_negative_qgwl_flag) then
+        local_negative_qgwl_sum = 0.0_r8 ! This will sum local NEGATIVE qgwl
+        local_positive_qgwl_sum = 0.0_r8 ! This will sum local POSITIVE qgwl
+
+        do nr = rtmCTL%begr, rtmCTL%endr
+            if (rtmCTL%qgwl(nr, nt_nliq) > 0.0_r8) then
+                local_positive_qgwl_sum = local_positive_qgwl_sum + rtmCTL%qgwl(nr, nt_nliq)
+            elseif (rtmCTL%qgwl(nr, nt_nliq) < 0.0_r8) then
+                local_negative_qgwl_sum = local_negative_qgwl_sum + rtmCTL%qgwl(nr, nt_nliq)
+            endif
+        enddo
+
+        ! Use reproducible sum for bit-for-bit reproducibility across PE layouts
+        block
+            real(r8) :: pos_local(1,1), pos_global(1)
+            real(r8) :: neg_local(1,1), neg_global(1)
+
+            ! Reproducible sum for positive qgwl
+            pos_local(1,1) = local_positive_qgwl_sum
+            call shr_reprosum_calc(pos_local, pos_global, 1, 1, 1, &
+                                   commid=mpicom_rof)
+            global_positive_qgwl_sum = pos_global(1)
+
+            ! Reproducible sum for negative qgwl
+            neg_local(1,1) = local_negative_qgwl_sum
+            call shr_reprosum_calc(neg_local, neg_global, 1, 1, 1, &
```

Comment:
You could combine the two calls to shr_reprosum_calc into one call because it looks like the two fields are independent of each other. That would be more efficient than two calls.